aten/src/ATen/native/Loss.cpp

```diff
@@ -79,7 +79,10 @@ Tensor margin_ranking_loss(const Tensor& input1, const Tensor& input2, const Ten
 Tensor kl_div(const Tensor& input, const Tensor& target, int64_t reduction) {
   auto output_pos = target * (at::log(target) - input);
   auto zeros = at::zeros_like(output_pos, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
+  // Perform a check for (target > -EPSILON) & (target < EPSILON) since floating point
+  // errors in output_pos cause the output values to be 1e-17 on GPU and -1e-14 on CPU.
   auto output = at::where(target > 0, output_pos, zeros);
```

Comment:
Does `zeros` have to be materialized? Can't it be expanded from a scalar tensor value?